### ISSUE TITLE
out_forward requires a user and password even when it's not needed

### ISSUE THREAD

**Issue by sxd** [fluent/fluent-bit#1612]
Configuring the out_forward to send logs to fluentd when fluentd use `auth = no` it requires `user` and `password` to be set in the configuration otherwise it's send this error

```
[2019/10/04 10:56:22] [error] [out_fw] server requires username and password
```

To workaround this issue it's necessary to set the data even with fake data

```
  user              user
  password    password
```

And with this it works and can send the data

The bug it's related to this commit https://github.com/fluent/fluent-bit/commit/8d6ea65328745868b23a220f9f7c8b6e8821604a

To reproduce just disable `auth` and unset `user` and `password`

**Comment by edsiper** [comment]
cc: @fujimotos

**Comment by edsiper** [comment]
@sxd please attach your fluent bit and fluentd configuration files.

**Comment by bluebike** [comment]
Our client had similar problem. 
If server sends HELO with **auth** option, username and password is required.
https://github.com/fluent/fluent-bit/blob/3d1456bdc60a327ab8a16b3658b8eab9ad46a7a9/plugins/out_forward/forward.c#L249

But in configuration (as far as I know) cannot set empty '' **username** and **password**.
Argument is mandatory for both.

Also fluentd seems to be setting HELO "auth" always (default empty string).

https://github.com/fluent/fluentd/blob/a14af3066bec35c7c2bd47e75953b3428a26ac00/lib/fluent/plugin/in_forward.rb#L422
So it forces for username and password.

Possbile Solutions 
1.  Assume username and password empty string 
2. If HELO.options.auth is empty string => no username, password
3. Have way to git empty username and password. 

To keep compatible with fluentd "empty" HELO.options.auth whould be consired as no username/password needed.  (right?)

**Comment by fujimotos** [comment]
Thanks @bluebike.

> To keep compatible with fluentd "empty" HELO.options.auth whould be consired as no username/password needed. (right?)

It turns out that Fluentd always includes non-empty auth token in _every_
HELO request. It's because the condition in in_forward.rb#L422 never
becomes false [due to the preceeding condition in line 201](https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/in_forward.rb#L201).

This means that we cannot tell whether username/password is required
by checking the `auth` field in the HELO request (because it's always there!)

From another angle, Fluentd's out_forward suggests that the right behaviour
here is to [use an empty string as the default value for username and password](https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/out_forward.rb#L139)
and generate a pong response based on it. 

Therefore, my conclusion is:

> 1. Assume username and password empty string

is the correct solution to apply here.

**Comment by fujimotos** [comment]
@edsiper I posted a patch that resolves the issue at #1616.

**Comment by edsiper** [comment]
thanks. The fix will be part of next 1.3.x release
